ledger/accountdb_test.go

```diff
@@ -730,6 +737,41 @@ func BenchmarkReadingAllBalances(b *testing.B) {
 	require.Equal(b, b.N, len(bal))
 }
 
+func BenchmarkReadingAllBalancesRAM(b *testing.B) {
+	benchmarkReadingAllBalances(b, true)
+}
+
+func BenchmarkReadingAllBalancesDisk(b *testing.B) {
+	benchmarkReadingAllBalances(b, false)
+}
+
+func benchmarkReadingRandomBalances(b *testing.B, inMemory bool) {
+	proto := config.Consensus[protocol.ConsensusCurrentVersion]
+	dbs, _ := dbOpenTest(b, inMemory)
+	setDbLogging(b, dbs)
+	defer dbs.close()
+
+	accounts := benchmarkInitBalances(b, dbs, proto)
+
+	qs, err := accountsDbInit(dbs.rdb.Handle, dbs.wdb.Handle)
+	require.NoError(b, err)
+
+	b.ResetTimer()
+	// read all the balances in the database, shuffled by map-iterator hash order
+	for addr, _ := range accounts {
+		_, _, err = qs.lookup(addr)
+		require.NoError(b, err)
+	}
```

Comment:
since this is a benchmark, you should wrap this portion in a for loop : 
```
for i:=0;i<b.N;i++ {
...
}
```
so that if it takes less than 1 second, go would try to execute it twice.

Reply:
nit: add something in the lines of 
```
b.ReportMetric(float64(len(accounts)), "accounts/iteration")
```
so that the benchmark feedback would be clear what is being tested.

Reply:
`b.N` goes into the number of accounts set up by `benchmarkInitBalances()`, so the loop to read back all the accounts is `b.N` long.